vm/src/builtins/descriptor.rs

```diff
@@ -0,0 +1,134 @@
+use rustpython_common::lock::PyRwLock;
+
+use super::{PyStr, PyType, PyTypeRef};
+use crate::class::PyClassImpl;
+use crate::function::Either;
+use crate::types::{Constructor, GetDescriptor, Unconstructible};
+use crate::{AsObject, Context, Py, PyObjectRef, PyPayload, PyRef, PyResult, VirtualMachine};
+
+#[derive(Debug)]
+pub struct DescrObject {
+    pub typ: PyTypeRef,
+    pub name: String,
+    pub qualname: PyRwLock<Option<String>>,
+}
+
+#[derive(Debug)]
+pub enum MemberKind {
+    ObjectEx = 16,
+}
+
+pub struct MemberDef {
+    pub name: String,
+    pub kind: MemberKind,
+    pub getter_or_offset: Either<fn(PyObjectRef, &VirtualMachine) -> PyResult, usize>,
```

Comment:
Using a dedicated enum type will be better to adding utility functions.

```rust
enum MemberGetter {
    Getter(fn(&VirtualMachine, PyObjectRef) -> PyResult),
    Offset(usize),
}
```
Swapping parameter orders in Getter will be helpful to have same convention with `PyGetterFnuc`.